### PR TITLE
Feature/ldap

### PR DESCRIPTION
**Pull request by SandroGreinecker**
Hallo Frano69! 
Hier der Code vom heutigen Pair-Programming.
Der Teamleiter und ich wünschen dir gute Besserung!

Pull request wurde abgesandt von mhlugr

### CODE REVIEW

File: Core.Ldap.Implementation/LdapClient.cs [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
+using System.Net;
 
 namespace Core.Ldap.Implementation;
 
 public class LdapClient : ILdapClient
 {
+    public SignInResult SignIn(SignInParams signInParams)
+    {
+        var credentials = new NetworkCredential(signInParams.Username, signInParams.Password, "HTL");
+        var serverId = new LdapDirectoryIdentifier("10.10.0.11");
```
Comment (EinboeckFranz) [inline Core.Ldap.Implementation/LdapClient.cs:12]: Local LDAP Server IP in global repository → Discuss with architecture team, if this should be changed. @htl-grieskirchen-core/architektur

File: Core.Ldap.Interface/SignInResult.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+namespace Core.Ldap.Interface;
+
+public class SignInResult
+{
+    public object User { get; set; } = null!;
```
Comment (EinboeckFranz) [inline Core.Ldap.Interface/SignInResult.cs:5]: User has object type, change before merging

**Review by EinboeckFranz — CHANGES_REQUESTED** [review]
2 changes required

File: Core.Ldap.Implementation/LdapClient.cs (hunk 1 above)
Comment (profiluefter) [reply Core.Ldap.Implementation/LdapClient.cs:12]: should be in appsettings.json injected via dependency injection
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-6.0#bind-hierarchical-configuration-data-using-the-options-pattern

Comment (SandroGreinecker) [reply Core.Ldap.Implementation/LdapClient.cs:12]: jo mochma

Comment (EinboeckFranz) [reply Core.Ldap.Implementation/LdapClient.cs]: Resolved in 751f1f63b768c07e61a77f29bd0ebe03532b413f

**Review by profiluefter — DISMISSED** [review]
Pfeifs!

File: Core.Ldap.Interface/LdapUser.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    /*
+     * The Mail address of the user
+     */
+    public string Email { get; set; } = null!;
+
+    /*
+     * One of three Organization Units the user could be part of
+     * e.g Schueler, Administrator, Lehrer
+     */
+    public string OrganizationUnit { get; set; } = null!;
```
Comment (profiluefter) [inline Core.Ldap.Interface/LdapUser.cs:30]: Could this also use the LdapGroup enum?

Comment (EinboeckFranz) [reply Core.Ldap.Interface/LdapUser.cs]: Implemented in e097d04e3748c6b0171e9e2aae3903ec41f005b3.

File: Core.Ldap.Interface/SignInResult.cs (hunk 2 above)
Comment (Klausmd5) [reply Core.Ldap.Interface/SignInResult.cs]: Outdated

**Review by profiluefter — DISMISSED** [review]

**Review by EinboeckFranz — COMMENTED** [review]
fixed

File: Core.Ldap.Implementation/LdapClient.cs [hunk 4]
```diff
@@ -1,7 +1,82 @@
...
+    private static LdapGroup GetOrganizationUnitOfString(string[] organizationUnits)
+    {
+        if (organizationUnits.Any(x => x.Contains("Administration")))
+            return LdapGroup.Administrator;
+        return organizationUnits.Any(x => x.Contains("Lehrer")) ? LdapGroup.Lehrer : LdapGroup.Schueler;
+    }
+
+    private static SearchResultEntryCollection GetDirectoryEntry(DirectoryConnection connection, string username)
+    {
+        var request = new SearchRequest("DC=htl,DC=grieskirchen,DC=local",
```
Comment (EinboeckFranz) [inline Core.Ldap.Implementation/LdapClient.cs:65]: Request throws exception on Linux, on Windows it somehow works

**Review by EinboeckFranz — CHANGES_REQUESTED** [review]
Find a solution that works on Linux as well

File: Core.Ldap.Implementation/LdapClient.cs (hunk 4 above)
Comment (EinboeckFranz) [reply Core.Ldap.Implementation/LdapClient.cs:68]: resolved in fcf6640db7b910966861bada79255f533c0b65c5

**Review by EinboeckFranz — APPROVED** [review]
works on both Linux and Windows

**Review by profiluefter — APPROVED** [review]

**Review by EinboeckFranz — APPROVED** [review]

**Review by Klausmd5 — APPROVED** [review]
whistles!

**Pull request merged**